app/assets/javascripts/controllers/toolbar_controller.js

```diff
@@ -0,0 +1,149 @@
+(function(){
+
+  /**
+  * Private method for subscribing to rxSubject.
+  * For success functuon @see ToolbarController#onRowSelect()
+  */
+  function subscribeToSubject() {
+    ManageIQ.angular.rxSubject.subscribe(function(event) {
+      if (event.rowSelect) {
+        this.onRowSelect(event.rowSelect);
+      }
+    }.bind(this),
+    function (err) {
+      console.error('Angular RxJs Error: ', err);
+    },
+    function () {
+      console.debug('Angular RxJs subject completed, no more events to catch.');
+    });
+  }
+
+  /**
+  * Private method for setting rootPoint of MiQEndpointsService.
+  * @param MiQEndpointsService service responsible for endpoits.
+  */
+  function initEndpoints(MiQEndpointsService) {
+    var urlPrefix = '/' + location.pathname.split('/')[1];
+    if (urlPrefix) {
+      MiQEndpointsService.rootPoint = urlPrefix;
+    }
+  }
+
+  function generateGetData(type, lastaction, display) {
+    var getData = {
+      id: ManageIQ.record.recordId,
+      gtl_type: type,
+      lastAction: lastaction,
+      display: display
+    }
+    return _.pick(getData, _.identity);
```

Comment:
This is weird :) are you using it to get a shallow clone or something? I'm sure there is a better method (besides, in lodash4, this will always return an empty object..)